src/data/mv_polynomial/rename.lean

```diff
@@ -243,4 +243,24 @@ end
 
 end coeff
 
+section support
+
+lemma support_rename_of_injective {p : mv_polynomial σ R} {f : σ → τ} (h : function.injective f) :
+  (rename f p).support = finset.image (map_domain f) p.support :=
+begin
+  rw finset.ext_iff,
+  intro a,
+  simp only [exists_prop, mem_support_iff, finset.mem_image, ne.def],
+  split,
+  { intro h1,
+    cases coeff_rename_ne_zero f p a h1 with d hd,
+    rw ← hd.1,
+    exact ⟨d, ⟨hd.2, by refl⟩⟩ },
+  { intro h,
+    cases h with b hb,
+    simpa only [←hb.right, coeff_rename_map_domain f h p b] using hb.left },
+end
```

Comment:
I'm pretty sure this is a special case of a much simpler statement. Can you prove the following instead:
```lean
lemma _root_.finsupp.map_domain_support_of_injective
  {α β M : Type*} [add_comm_monoid M] [decidable_eq β] {f : α → β} (hf : function.injective f)
  (s : α →₀ M) :
  (map_domain f s).support = finset.image f s.support :=
sorry
```
which makes this
```suggestion
lemma support_rename_of_injective {p : mv_polynomial σ R} {f : σ → τ} (h : function.injective f) :
  (rename f p).support = finset.image (map_domain f) p.support :=
begin
  rw rename_eq,
  exact finsupp.map_domain_support_of_injective (map_domain_injective h) _,
end
```
Obviously `finsupp.map_domain_support_of_injective` belongs in another file; right next to `finsupp.map_domain_support` would be good.

Reply:
Here's one proof of `map_domain_support_of_injective`.
```lean
finset.subset.antisymm finsupp.map_domain_support $ begin
  rw finset.image_subset_iff_subset_preimage (hf.inj_on _),
  intros x hx,
  simp[map_domain_apply hf, finsupp.mem_support_iff.mp hx],
end
```
but there might be a more direct one

Reply:
Nice! I'll use your proof, the proof I came up with is more complicated:

```
begin
  rw finset.ext_iff,
  intro b,
  split,
  { exact λ h, map_domain_support h },
  { simp only [exists_prop, mem_support_iff, finset.mem_image, ne.def],
    intro h,
    cases h with a h,
    simpa [← h.2, map_domain_apply hf] using h.1 },
end
```